Tella/Scenes/Audio/AudioPlayer/ViewModel/AudioPlayerViewModel.swift

```diff
@@ -5,52 +5,52 @@
 import Foundation
 import Combine
 
-enum PlayState {
-    case ready
-    case playing
-}
 
 class AudioPlayerViewModel: ObservableObject {
     
-    @Published var state: PlayState = .ready
+    @Published var isPlaying = false
     
     @Published var shouldDisableFastForwardButton : Bool = false
     @Published var shouldDisableRewindBackButton : Bool = false
 
     @Published var currentTime : String  = "00:00:00"
     @Published var duration : String  = "00:00:00"
 
-    private var cancellable: Set<AnyCancellable> = []
-
+    var mainAppModel: MainAppModel
+    var cancellable: Set<AnyCancellable> = []
+    var currentFile: VaultFileDB?
     var audioPlayerManager: AudioPlayerManager = AudioPlayerManager()
     
      var currentData : Data?
   
     @Published var audioIsReady = false
 
-    init(currentData: Data?) {
-         self.currentData = currentData
-
+    @Published var timeDuration: TimeInterval?
+    
+    init(currentFile: VaultFileDB?, mainAppModel: MainAppModel) {
+        self.currentFile = currentFile
+        self.mainAppModel = mainAppModel
+        if let currentFile {
+            self.currentData = self.mainAppModel.vaultManager.loadFileData(file: currentFile)
+        }
         audioPlayerManager.audioPlayer.currentTime.sink { value in
-            self.currentTime = value.stringFromTimeInterval()
+            self.currentTime = value.toHHMMSSString()
         }.store(in: &self.cancellable)
         
         audioPlayerManager.audioPlayer.duration.sink { value in
-            self.duration = value.stringFromTimeInterval()
+            self.duration = value.toHHMMSSString()
         }.store(in: &self.cancellable)
         
         audioPlayerManager.audioPlayer.audioPlayerDidFinishPlaying.sink { [self] value in
-            self.state = .ready
-            
+            self.isPlaying = false
             self.shouldDisableFastForwardButton = true
```

Comment:
I suggest to move this code to a separate function beacause it's duplicated in `onPausePlaying()`
```
            self.isPlaying = false
            self.shouldDisableFastForwardButton = true
            self.shouldDisableRewindBackButton = true
            self.updateView()
 

```